administrator/components/com_categories/src/Model/CategoriesModel.php

```diff
@@ -239,8 +241,31 @@ protected function getListQuery()
                 ->bind(':extension', $extension);
         }
 
-        // Filter on the level.
-        if ($level = (int) $this->getState('filter.level')) {
+        // Filter by categories and by level
+        $categoryId = $this->getState('filter.category_id', []);
+        $level      = $this->getState('filter.level');
+
+        if (!is_array($categoryId)) {
+            $categoryId = $categoryId ? [$categoryId] : [];
+        }
+
+        // Case: Using both categories filter and by level filter
+        if (count($categoryId)) {
+            $categoryTable    = Table::getInstance('Category', 'JTable');
+            $subCatItemsWhere = [];
+
+            foreach ($categoryId as $filter_catid) {
+                $categoryTable->load($filter_catid);
```

Comment:
```suggestion
            foreach ($categoryId as $filterCatId) {
                $categoryTable->load($filterCatId);
```
Only fixing snake_case verses camelCase